test/devicecontroller/DefaultDeviceController.test.ts

```diff
@@ -1037,25 +1035,300 @@ describe('DefaultDeviceController', () => {
+
+      deviceController.chooseVideoInputDevice(device).then(() => {});
+
+      new TimeoutScheduler(200).start(() => {
+        deviceController.chooseVideoInputDevice(device);
+      });
+
+      new TimeoutScheduler(250).start(() => {
+        deviceController.chooseVideoInputDevice(null);
+      });
+
+      new TimeoutScheduler(300).start(() => {
+        device.stop();
+        done();
+      });
+    });
+
+    it('can replace the local video for VideoTransformDevice', async () => {
+      domMockBehavior.getUserMediaSucceeds = true;
+      const mockVideoStream = new MediaStream();
+      // @ts-ignore
+      mockVideoStream.id = 'sample';
+      // @ts-ignore
+      mockVideoStream.active = true;
+      // @ts-ignore
+      const mockVideoTrack = new MediaStreamTrack('test', 'video');
+      mockVideoStream.addTrack(mockVideoTrack);
+      domMockBehavior.createElementCaptureStream = mockVideoStream;
+
+      deviceController.chooseVideoInputDevice(stringDeviceId).then(() => {});
+      const processor = new NoOpVideoFrameProcessor();
+      const device = new DefaultVideoTransformDevice(logger, stringDeviceId, [processor]);
+      let called = false;
+      class TestAudioVideoController extends NoOpAudioVideoController {
+        async replaceLocalVideo(): Promise<void> {
+          called = true;
+        }
+      }
+
+      audioVideoController = new TestAudioVideoController();
+      deviceController.bindToAudioVideoController(audioVideoController);
+      audioVideoController.videoTileController.startLocalVideoTile();
+
+      await deviceController.chooseVideoInputDevice(device);
+
+      const stream = await deviceController.acquireVideoInputStream();
+      deviceController.releaseMediaStream(stream);
+      await new Promise(resolve => setTimeout(resolve, 200));
+      await deviceController.chooseVideoInputDevice(null);
+
+      expect(called).to.be.true;
+      await device.stop();
+    });
+
+    it('can choose media stream as device then choose VideoTransformDevice without audio video controller', done => {
+      domMockBehavior.getUserMediaSucceeds = true;
+      const mockVideoStream = new MediaStream();
+      // @ts-ignore
+      mockVideoStream.active = true;
+      // @ts-ignore
+      mockVideoStream.id = 'sample';
+      // @ts-ignore
+      const mockVideoTrack = new MediaStreamTrack('test', 'video');
+      mockVideoStream.addTrack(mockVideoTrack);
+      domMockBehavior.createElementCaptureStream = mockVideoStream;
+      deviceController.chooseVideoInputDevice(mockVideoStream).then(() => {});
+      const processor = new NoOpVideoFrameProcessor();
+      const device = new DefaultVideoTransformDevice(logger, mockVideoStream, [processor]);
+      deviceController.chooseVideoInputDevice(device).then(() => {});
+      new TimeoutScheduler(400).start(() => {
+        deviceController.chooseVideoInputDevice(null);
+      });
```

Comment:
I don't get it. Are you trying to `await choose(); await choose(null); await device.stop()`?

Reply:
removed the timeout scheduler